### PR TITLE
Update .NET Framework GC config info

### LINKED ISSUE #9739: Not clear what the default mode for Server GC is
.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1fcebd46-8a64-9531-f6e1-3a330b61bb0f
* Version Independent ID: 7538a252-8fa8-4615-92df-f1c6d7d6a3e4
* Content: [Latency Modes](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/latency)
* Content Source: [docs/standard/garbage-collection/latency.md](https://github.com/dotnet/docs/blob/master/docs/standard/garbage-collection/latency.md)
* Product: **dotnet**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### LINKED ISSUE #11146: Document additional GC configuration settings
## Document additional GC configuration settings

A number of GC configuration settings are currently undocumented but should be documented. They include:

- GCHeapCount
- GCNoAffinitize
- GCHeapAffinitizeMask

They can be added to the [Fundamentals of Garbage Collection](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/fundamentals#workstation-and-server-garbage-collection) article.

//cc @Maoni0

### PR DESCRIPTION
**Pull request by rpetrusha**
This PR adds 3 GC-related .NET Framework runtime configuration settings.

Fixes #11146

(The .NET Core config options will be handled in a separate PR.)

### CODE REVIEW

File: docs/core/runtime-config/envvars.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+# Runtime configuration with environment variables
+
+.NET Core supports a number of environment variables that you can use to configure your applications. 
+
+The following sections list selected supported environment variables by category. For a complete list of supported, as well as unsupported and internal environment variables, see [Host Configuration Knobs](https://github.com/dotnet/coreclr/blob/a28b25aacdcd2adb0fdfa70bd869f53ba6565976/Documentation/project-docs/clr-configuration-knobs.md).
+
+> [!NOTE]
+> This documentation is a work in progress. If you notice that the information presented here is either incomplete or inaccurate, either [open an issue](https://github.com/dotnet/docs/issues) to let us know about it, or [submit a pull request](https://github.com/dotnet/docs/pulls) to add or correct the information. For information on submitting pull requests for the dotnet/docs repo, see the [contributor's guide](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
+
+## Application domains
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: We should not be documenting any of these.

File: docs/core/runtime-config/envvars.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPlus_ADULazyMemoryRelease`|0 or 1|Determines whether a lazy memory release occurs when an app domain is unloaded. It is on (1) by default; it should be turned off (0) to catch memory leaks, in which case the unload should be immediately followed by a garbage collection.|
+|`COMPlus_ADURetryCount`|*n*|Controls the timeout of an app domain unload. It is used for workarounds when a machine is too slow, there are network issues, etc.|
+|`COMPlus_FinalizeOnShutdown`|0 or 1|When enabled, on shutdown, it blocks all user threads and calls finalizers for all finalizable objects, including live objects.|
+
+## Assembly loading
+
+|Variable name|Variable value|Description|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: This is a very special setting for Visual Studio designer and nobody else. We should not be documenting it.

File: docs/core/runtime-config/envvars.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+## Debugging
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPlus_DbgPackShimPath`|*string_path*|Defines the CoreCLR path to dbgshim.dll. Support for this environment variable may be removed in the future.|
+|`COMPlus_DbgRedirectApplication`|*string_path*|Specifies the auxiliary debugger application to launch.|
+|`COMPlus_DbgRedirectAttachCmd`|*string_params*|Specifies command parameters for attaching the auxiliary debugger.|
+|`COMPlus_DbgRedirectCommonCmd`|*string_format*|Specifies a command line format string for the auxiliary debugger.|
+|`COMPlus_DbgRedirectCreateCmd`|*string_params*|Specifies command parameters when creating the auxiliary debugger.|
+|`COMPLUS_EnableDiagnostics`|0 or 1|Enables (1) or disables (0) debugger and profiler diagnostics.|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: The casing for this one and number of others is off. It should be `COMPlus`

File: docs/core/runtime-config/envvars.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+## Assembly loading
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|COMPlus_designerNamespaceResolution|0 or 1|Enables (1) or disables (0) the <xref:System.Runtime.InteropServices.WindowsRuntime.WindowsRuntimeMetadata.DesignerNamespaceResolve?displayProperty=nameWithType> event for WinRT types.|
+
+## Debugging
+
+|Variable name|Variable value|Description|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: @noahfalk Do you expect that these switches should be documented?

File: docs/core/runtime-config/envvars.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+|`COMPLUS_GCCpuGroup`|0 or 1|Enables (1) or disables (0, the default value) support for CPU groups by the garbage collector if server garbage collection is enabled.|
+|`COMPLUS_GCLatencyLevel`|see <xref:System.Runtime.GCLatencyMode>|Adjusts the intrusiveness of the garbage collector. The default value is 1 (<xref:System.Runtime.GCLatencyLevel.Interactive?displayProperty=nameWithType>). For more information, see [Latency modes](../../standard/garbage-collection/latency.md).
+|`COMPLUS_GCName`|*string_path*|Specifies a path to the library containing the GC that the runtime intends to load. For more information, see [Standalone GC Loader Design](https://github.com/dotnet/coreclr/blob/4c7c066d0bacdb86a2311333de1ca73d94ae5280/Documentation/design-docs/standalone-gc-loading.md).|
+|`COMPLUS_GCPollType`|0 or 1|Enables (1) or disables (0, the default value) [`JIT_PollGC`](https://github.com/dotnet/coreclr/blob/deb00ad58acf627763b6c0a7833fa789e3bb1cd0/src/vm/jithelpers.cpp#L6331-L6536) calls to be inserted into user code.|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: `COMPLUS_GCPollType` should not be documented.

File: docs/core/runtime-config/envvars.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+|`COMPLUS_GCStressStart`|*n*|Start GCStress after *n* stress GCs have been attempted. The default value is 0.|
+|`COMPLUS_gcTrimCommitOnLowMemory`|0 or 1|When enabled (1), aggressively trims the committed memory for the ephemeral segment to keep as little memory committed as possible. The default is off (0).|
+
+## Interop
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`AllowDComReflection`|0 or 1|Enables (1) or disables (0, the default value) the ability to marshal blocked reflection types by out-of-process DCOM clients.|
+|`InteropLogArguments`|Enables (1) or disables (0, the default value) logging of all pinned arguments passed to an interop call.|
+|`PreferComInsteadOfManagedRemoting`|0 or 1|Determines whether COM (1) or managed remoting (0, the default value) is used to communicate with a cross-app domain COM callable wrapper.|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: Removing does not exist in CoreCLR

File: docs/core/runtime-config/envvars.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+## JIT compiler
+
+|Variable name|Variable value|Description|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: None of these should be documented. They do not even work in release runtime.

File: docs/core/runtime-config/envvars.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+|--|--|--|
+|`COMPLUS_JitDump`|*string-method*|Dumps trees for the method named *string-method*.|
+|`COMPLUS_JitNoRngChks`|0 or 1|Enables (0) or disables (1) range checks.|
+
+## Loader
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPLUS_ForceLog`|Enables (1) or disables (0, the default value) the Fusion assembly binding log. The log can be viewed with the [Assembly Binding Log Viewer](https://docs.microsoft.com/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer).|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: There is no fusion in .NET Core. This logging is not of much use in .NET Core.

File: docs/core/runtime-config/envvars.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+description: "Use environment variables to configure the runtime behavior of .NET Core applications."
+ms.date: "06/19/2019"
+author: "rpetrusha"
+ms.author: "ronpet"
+---
+# Runtime configuration with environment variables
+
+.NET Core supports a number of environment variables that you can use to configure your applications. 
+
+The following sections list selected supported environment variables by category. For a complete list of supported, as well as unsupported and internal environment variables, see [Host Configuration Knobs](https://github.com/dotnet/coreclr/blob/a28b25aacdcd2adb0fdfa70bd869f53ba6565976/Documentation/project-docs/clr-configuration-knobs.md).
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: You should confirm with each of feature team that they are comfortable officially supporting each of these variables. I have commented on the obvious ones that we should not be documenting, but there is likely a bunch more.

A simple rule should be: All officially supported variables should have tests. If we do not have a test for it, we should not be declaring it as officially supported.

File: docs/core/runtime-config/envvars.md (hunk 4 above)
Comment (noahfalk) [reply docs/core/runtime-config/envvars.md]: I would only expect COMPLUS_EnableDiagnostics to be documented in this section, none of the others.

cc @mikem8361 who I think added EnableDiagnostics

Comment (mikem8361) [reply docs/core/runtime-config/envvars.md]: Looks fine to me.

File: docs/core/runtime-config/envvars.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+|`COMPLUS_GCLatencyLevel`|see <xref:System.Runtime.GCLatencyMode>|Adjusts the intrusiveness of the garbage collector. The default value is 1 (<xref:System.Runtime.GCLatencyLevel.Interactive?displayProperty=nameWithType>). For more information, see [Latency modes](../../standard/garbage-collection/latency.md).
+|`COMPLUS_GCName`|*string_path*|Specifies a path to the library containing the GC that the runtime intends to load. For more information, see [Standalone GC Loader Design](https://github.com/dotnet/coreclr/blob/4c7c066d0bacdb86a2311333de1ca73d94ae5280/Documentation/design-docs/standalone-gc-loading.md).|
+|`COMPLUS_GCPollType`|0 or 1|Enables (1) or disables (0, the default value) [`JIT_PollGC`](https://github.com/dotnet/coreclr/blob/deb00ad58acf627763b6c0a7833fa789e3bb1cd0/src/vm/jithelpers.cpp#L6331-L6536) calls to be inserted into user code.|
+|`COMPLUS_GCStress`|0 or 1|If on (1), triggers garbage collections at regular intervals. The default is off (0).|
+|`COMPLUS_GCStressStart`|*n*|Start GCStress after *n* stress GCs have been attempted. The default value is 0.|
```
Comment (Maoni0) [inline docs/core/runtime-config/envvars.md]: agree with Jan, GCPollType should be not public. 
GCStress/GCStressStart are actually not GC config. they are enabled to help debug managed heap corruption.

File: docs/core/runtime-config/envvars.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+|`COMPLUS_gcTrimCommitOnLowMemory`|0 or 1|When enabled (1), aggressively trims the committed memory for the ephemeral segment to keep as little memory committed as possible. The default is off (0).|
```
Comment (Maoni0) [inline docs/core/runtime-config/envvars.md]: I know we already have this documented (someone asked to document without consulting the GC team); this should be not public to begin with. this is not handled in core. I should just remove this from the code.

**Comment by Maoni0** [comment]
if we are going to document COMPlus_GCCpuGroup we should also document COMPlus_Thread_UseAllCpuGroups

**Comment by mairaw** [comment]
There are some conflicts here @rpetrusha. Do you wanna take a look?

**Comment by davidwrighton** [comment]
@aaronrobinsonmsft could you look at the environment variables around interop that @rpetrusha is documenting here? I suspect neither of these should be documented.

File: docs/core/runtime-config/envvars.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+## Garbage collection
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPlus_gcAllowVeryLargeObjects`|0 or 1|Enables (1, the default value) or Disables (0) support on 64-bit platforms for arrays that are greater than 2 gigabytes (GB) in total size.|
+|`COMPlus_GCCpuGroup`|0 or 1|Enables (1) or disables (0, the default value) support for CPU groups by the garbage collector if server garbage collection is enabled.|
+|`COMPlus_GCLatencyLevel`|see <xref:System.Runtime.GCLatencyMode>|Adjusts the intrusiveness of the garbage collector. The default value is 1 (<xref:System.Runtime.GCLatencyMode.Interactive?displayProperty=nameWithType>). For more information, see [Latency modes](../../standard/garbage-collection/latency.md).
+|`COMPlus_GCName`|*string_path*|Specifies a path to the library containing the GC that the runtime intends to load. For more information, see [Standalone GC Loader Design](https://github.com/dotnet/coreclr/blob/4c7c066d0bacdb86a2311333de1ca73d94ae5280/Documentation/design-docs/standalone-gc-loading.md).|
+
+## Interop
```
Comment (AaronRobinsonMSFT) [inline docs/core/runtime-config/envvars.md]: I would not document either of these "Interop" variables.

**Review by davidwrighton — APPROVED** [review]
LGTM

**Comment by mairaw** [comment]
There are new merge conflicts here @rpetrusha. Do you want to take a look?

**Comment by rpetrusha** [comment]
I've resolved the merge conflicts, @mairaw.

File: docs/core/runtime-config/envvars.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+## Garbage collection
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`COMPlus_gcAllowVeryLargeObjects`|0 or 1|Enables (1, the default value) or Disables (0) support on 64-bit platforms for arrays that are greater than 2 gigabytes (GB) in total size.|
+|`COMPlus_GCCpuGroup`|0 or 1|Enables (1) or disables (0, the default value) support for CPU groups by the garbage collector if server garbage collection is enabled.|
+|`COMPlus_GCLatencyLevel`|see <xref:System.Runtime.GCLatencyMode>|Adjusts the intrusiveness of the garbage collector. The default value is 1 (<xref:System.Runtime.GCLatencyMode.Interactive?displayProperty=nameWithType>). For more information, see [Latency modes](../../standard/garbage-collection/latency.md).
+|`COMPlus_GCName`|*string_path*|Specifies a path to the library containing the GC that the runtime intends to load. For more information, see [Standalone GC Loader Design](https://github.com/dotnet/coreclr/blob/4c7c066d0bacdb86a2311333de1ca73d94ae5280/Documentation/design-docs/standalone-gc-loading.md).|
+
+## Networking
```
Comment (Youssef1313) [inline docs/core/runtime-config/envvars.md]:
```suggestion
## Networking

```

File: docs/core/runtime-config/runtimeconfig.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+# Runtime configuration with runtimeconfig.json
+
+You can use the **configProperties** section of the *runtimeconfig.json* file to set designated properties at runtime. 
+
+> [!NOTE]
+> This documentation is a work in progress. If you notice that the information presented here is either incomplete or inaccurate, either [open an issue](https://github.com/dotnet/docs/issues) to let us know about it, or [submit a pull request](https://github.com/dotnet/docs/pulls) to add or correct the information. For information on submitting pull requests for the dotnet/docs repo, see the [contributor's guide](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
+
+## File format
+
+The  *runtimeconfig.json* file has the following format:
```
Comment (Youssef1313) [inline docs/core/runtime-config/runtimeconfig.md]:
```suggestion
The *runtimeconfig.json* file has the following format:
```

File: docs/core/runtime-config/runtimeconfig.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+         "config-property-name1": config-value1,
+         "config-property-name2": config-value2
+      }
+   }
+}
+```
+
+## Individual configuration settings
+
+The  *runtimeconfig.json* file supports the following configuration settings:
```
Comment (Youssef1313) [inline docs/core/runtime-config/runtimeconfig.md]:
```suggestion
The *runtimeconfig.json* file supports the following configuration settings:
```

File: docs/framework/configure-apps/file-schema/runtime/gcheapaffinitizemask-element.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+  
+## Syntax  
+  
+```xml  
+<GCHeapAffinitizeMask   
+   enabled="nnnn"/>  
+```  
+  
+## Attributes and elements  
+ The following sections describe attributes, child elements, and parent elements.
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcheapaffinitizemask-element.md]:
```suggestion

The following sections describe attributes, child elements, and parent elements.  
```

File: docs/framework/configure-apps/file-schema/runtime/gcheapaffinitizemask-element.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+</configuration>  
+```  
+
+## See also
+
+- <xref:System.Runtime.GCSettings.IsServerGC%2A?displayProperty=nameWithType>
+- [\<GCNoAffinitize> element](gcnoaffinitize-element.md) element
+- [\<GCHeapCount> element](gcheapcount-element.md)
+- [Fundamentals of garbage collection](../../../../standard/garbage-collection/fundamentals.md)
+- [Runtime Settings Schema](../../../../../docs/framework/configure-apps/file-schema/runtime/index.md)
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcheapaffinitizemask-element.md]:
```suggestion
- [Runtime Settings Schema](index.md)
```

File: docs/framework/configure-apps/file-schema/runtime/gcheapaffinitizemask-element.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+- [Configuration File Schema](../../../../../docs/framework/configure-apps/file-schema/index.md)
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcheapaffinitizemask-element.md]:
```suggestion
- [Configuration File Schema](../index.md)
```

File: docs/framework/configure-apps/file-schema/runtime/gcheapcount-element.md [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+  
+## Syntax  
+  
+```xml  
+<GCHeapCount    
+   enabled="nn"/>  
+```  
+  
+## Attributes and elements  
+ The following sections describe attributes, child elements, and parent elements.
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcheapcount-element.md]:
```suggestion

The following sections describe attributes, child elements, and parent elements.
```

File: docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md [hunk 20]
```diff
@@ -0,0 +1,91 @@
...
+title: "<GCNoAffinitize> element"
+ms.date: "06/19/2019"
+helpviewer_keywords: 
+  - "gcNoAffinitize element"
+  - "<gcNoAffinitize> element"
+author: "rpetrusha"
+ms.author: "ronpet"
+---
+# \<GCNoAffinitize> element
+Specifies whether or not to affinitize server GC threads with CPUs.
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md]:
```suggestion

Specifies whether or not to affinitize server GC threads with CPUs.
```

File: docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+  
+## Syntax  
+  
+```xml  
+<GCNoAffinitize    
+   enabled="true|false"/>  
+```  
+  
+## Attributes and elements  
+ The following sections describe attributes, child elements, and parent elements.
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md]:
```suggestion

The following sections describe attributes, child elements, and parent elements.
```

File: docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+```xml
+<configuration>  
+   <runtime>  
+      <gcServer enabled=”true”/>
+      <GCNoAffinitize enabled=”true”/>   </runtime>  
+   </runtime>
+</configuration>  
+```
+
+The following example does not affinitize server GC threads and limits the number of GC heaps/threads to 10.
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md]:
```suggestion
The following example does not affinitize server GC threads and limits the number of GC heaps/threads to 10:
```

File: docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+</configuration>  
+```
+
+## See also
+
+- <xref:System.Runtime.GCSettings.IsServerGC%2A?displayProperty=nameWithType>
+- [\<GCHeapAffinitizeMask> element](gcheapaffinitizemask-element.md)
+- [\<GCHeapCount> element](gcheapcount-element.md) element
+- [Fundamentals of garbage collection](../../../../standard/garbage-collection/fundamentals.md)
+- [Runtime Settings Schema](../../../../../docs/framework/configure-apps/file-schema/runtime/index.md)
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md]:
```suggestion
- [Runtime Settings Schema](index.md)
```

File: docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+- [Configuration File Schema](../../../../../docs/framework/configure-apps/file-schema/index.md)
```
Comment (Youssef1313) [inline docs/framework/configure-apps/file-schema/runtime/gcnoaffinitize-element.md]:
```suggestion
- [Configuration File Schema](../index.md)
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

**Comment by mairaw** [comment]
Resolved some merge conflicts

File: docs/core/runtime-config/envvars.md (hunk 9 above)
Comment (mairaw) [reply docs/core/runtime-config/envvars.md]: Should this point to master?

```suggestion
The following sections list selected supported environment variables by category. For a complete list of supported, as well as unsupported and internal environment variables, see [Host Configuration Knobs](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md).
```

File: docs/core/runtime-config/index.md [hunk 25]
```diff
@@ -0,0 +1,58 @@
...
+ 
+.NET Core provides the following mechanisms for configuring applications at runtime:
+
+- The **configProperties** section of the *runtimeconfig.json* file. This section has the form:
+
+   ```json
+   {
+      "runtimeOptions": {
+         "configProperties": {
+            "config-property-name1": config-value1,
```
Comment (mairaw) [inline docs/core/runtime-config/index.md]: should we have these values in quotes to not cause this red sea below? 😄 

```suggestion
            "config-property-name1": "config-value1",
```

**Review by gewarren — DISMISSED** [review]

**Comment by mairaw** [comment]
@jkotas is this one good to merge or do we still need approval from other teams about talking about those env variables?

File: docs/core/runtime-config/envvars.md [hunk 26]
```diff
@@ -0,0 +1,63 @@
+---
+title: "Runtime configuration with environment variables"
+description: "Use environment variables to configure the runtime behavior of .NET Core applications."
+ms.date: "06/19/2019"
+---
+# Runtime configuration with environment variables
+
+.NET Core supports a number of environment variables that you can use to configure your applications. 
+
+The following sections list selected supported environment variables by category. For a complete list of supported, as well as unsupported and internal environment variables, see [Host Configuration Knobs](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md).
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]:
```suggestion
The following sections list selected supported environment variables by category. For a complete list of supported, as well as unsupported and internal environment variables, see [Configuration Knobs](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/clr-configuration-knobs.md).
```

File: docs/core/runtime-config/envvars.md [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+|`COMPlus_GCLatencyLevel`|see <xref:System.Runtime.GCLatencyMode>|Adjusts the intrusiveness of the garbage collector. The default value is 1 (<xref:System.Runtime.GCLatencyMode.Interactive?displayProperty=nameWithType>). For more information, see [Latency modes](../../standard/garbage-collection/latency.md).
+|`COMPlus_GCName`|*string_path*|Specifies a path to the library containing the GC that the runtime intends to load. For more information, see [Standalone GC Loader Design](https://github.com/dotnet/coreclr/blob/4c7c066d0bacdb86a2311333de1ca73d94ae5280/Documentation/design-docs/standalone-gc-loading.md).|
+
+## Networking
+
+|Variable name|Variable value|Description|
+|--|--|--|
+|`DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_HTTP2SUPPORT`|0 or 1|Starting with .NET Core 3.0, indicates whether support for the HTTP/2 protocol is enabled (1) or disabled (0, the default value).|
+|`DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER`|0 or 1|Enables the use of <xref:System.Net.Http.SocketsHttpHandler?displayProperty=nameWithType> (1, the default value) or of the legacy <xref:System.Net.Http.HttpClientHandler?displayProperty=nameWithType> (0) with high-level networking APIs.|
```
Comment (jkotas) [inline docs/core/runtime-config/envvars.md]: HttpClientHandler is not legacy. It is abstraction that covers both cases. 

`0` enables HttpClientHandler implementation based on [libcurl](https://curl.haxx.se/libcurl/). The name of this implementation is CurlHandler, but it is internal name, not exposed publicly.

File: docs/core/runtime-config/runtimeconfig.md [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+|"System.Net.Http.SocketsHttpHandler.Http2Support"|true &#124; false|Determines whether high-level networking APIs such as <xref:System.Net.Http.HttpClient> use <xref:System.Net.Http.SocketsHttpHandler> (true, the default value) or the legacy >xref:System.Net.Http.HttpClientHandler> (false).<br/><br/>You can also configure this setting programmatically by calling the <xref:System.AppContext.SetSwitch%2A?displayProperty=nameWithType> method.|
+|"System.Net.Http.UseSocketsHttpHandler"|true &#124; false|Determines whether high-level networking APIs such as <xref:System.Net.Http.HttpClient?displayProperty=nameWithType> use <xref:System.Net.Http.SocketsHttpHandler?displayProperty=nameWithType> (`true`, the default value) or <xref:System.Net.Http.HttpClientHandler?displayProperty=nameWithType> (`false`).<br/><br/>You can also configure this setting programmatically by calling the <xref:System.AppContext.SetSwitch%2A?displayProperty=nameWithType> method.|
```
Comment (jkotas) [inline docs/core/runtime-config/runtimeconfig.md]: Dtto

**Comment by jkotas** [comment]
> do we still need approval from other teams about talking about those env variables?

Yes, I would like @Maoni0 or somebody else from the GC team to sign-off on the GC-specific content (the majority of this PR).

**Comment by jkotas** [comment]
The non-GC specific content looks good to me, modulo comments.

File: docs/core/runtime-config/runtimeconfig.md [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+|"System.GC.Concurrent"|true &#124; false|Determines whether background garbage collection is enabled (`true`, the default value) or disabled (`false`). For more information, see [Background workstation garbage collection](../../standard/garbage-collection/fundamentals.md#background-workstation-garbage-collection) and [Background server garbage collection](../../standard/garbage-collection/fundamentals.md#background-server-garbage-collection).|
+|"System.GC.RetainVM"|true &#124; false|Determines whether segments that should be deleted are put on a standby list for future use (`true`) or are released back to the operating system (`false`, the default value).|
+|"System.GC.Server"|true &#124; false|Determines whether the application uses server garbage collection (`true`) or workstation garbage collection (`false`, the default value). For more information, see [Configuring garbage collection](../../standard/garbage-collection/fundamentals.md#configuring-garbage-collection).|
```
Comment (Maoni0) [inline docs/core/runtime-config/runtimeconfig.md]: are these supposed to be an exhaustive list of what's supported in runtimeconfig.json in 3.0?

**Comment by Maoni0** [comment]
I started to look but had a question what the scope is; is this supposed to cover all that's supported in .net core 3.0? 'cause I see that there are only 3 System.GC configs in runtimeconfig.json.

CC @andy-ms.

File: docs/core/runtime-config/runtimeconfig.md (hunk 29 above)
Comment (mairaw) [reply docs/core/runtime-config/runtimeconfig.md]: I believe @rpetrusha was trying to document all of them.

Comment (Maoni0) [reply docs/core/runtime-config/runtimeconfig.md]: I see, then if you do a search for "System.GC" in eeconfig.cpp, everything you see there should be documented.

**Review by gewarren — APPROVED** [review]

**Pull request merged**